src/mono/msbuild/apple/build/AppleBuild.LocalBuild.props

```diff
@@ -56,6 +57,15 @@
     <LibraryBuilderDir>$([MSBuild]::NormalizeDirectory($(BuildBaseDir), 'LibraryBuilder'))</LibraryBuilderDir>
 
     <MonoAotCrossDir>$([MSBuild]::NormalizePath($(BuildBaseDir), 'cross'))</MonoAotCrossDir>
+    <CoreCLRILCompilerDir>$([MSBuild]::NormalizeDirectory('$(BuildBaseDir)', 'ilc'))</CoreCLRILCompilerDir>
+    <ILCompilerTargetsDir>$([MSBuild]::NormalizePath('$(BuildBaseDir)', 'BuildIntegration'))</ILCompilerTargetsDir>
+    <ILCompilerTargetsPath>$([MSBuild]::NormalizePath('$(ILCompilerTargetsDir)', 'Microsoft.DotNet.ILCompiler.SingleEntry.targets'))</ILCompilerTargetsPath>
+    <IlcToolsPath>$(CoreCLRILCompilerDir)</IlcToolsPath>
+    <IlcSdkPath>$([MSBuild]::NormalizeDirectory('$(BuildBaseDir)', 'aotsdk'))</IlcSdkPath>
+    <IlcFrameworkPath>$(MicrosoftNetCoreAppRuntimePackLibDir)</IlcFrameworkPath>
+    <IlcFrameworkNativePath>$(MicrosoftNetCoreAppRuntimePackNativeDir)</IlcFrameworkNativePath>
+    <ILLinkTargetsPath>$([MSBuild]::NormalizePath('$(BuildBaseDir)', 'microsoft.net.illink.tasks', 'build', 'Microsoft.NET.ILLink.targets'))</ILLinkTargetsPath>
+    <ILLinkTasksAssembly>$([MSBuild]::NormalizePath('$(BuildBaseDir)', 'microsoft.net.illink.tasks', 'tools', '$(_NetCoreAppCurrent)', 'ILLink.Tasks.dll'))</ILLinkTasksAssembly>
```

Comment:
Why on: https://github.com/dotnet/runtime/pull/87773/files#diff-b20fc29acfd5924f74eaf71e6cafeea421cb1fc220a078fff9d948927dca1531R61 we take `microsoft.net.illink.tasks` from restored NuGet packages and here from the built dir?

Reply:
This is how AOT compilation on Helix works. On a build machine, the runtime is built, and an app payload for AOT compilation is created. When sending data to Helix, we include both the payload and the dependencies required for AOT compilation since the Helix machine doesn't have any runtime bits.

On a build machine, a path from the runtime is included. On a Helix machine, `BuildBaseDir` includes all dependencies required for the AOT compilation. Here is the mapping: https://github.com/dotnet/runtime/blob/728e32b93e3acf57f6c07e9ad6e40409d56138a1/src/libraries/sendtohelix-mobile.targets#L63-L72

Reply:
Thanks for the explanation.